File: api/settlement/domain/settlement.go
package domain

import "github.com/gofrs/uuid/v5"

type Settlement struct {
	ID                  uuid.UUID `json:"id"`
	Name                string    `json:"name"`
	Owner               string    `json:"owner"`
	SurvivalLimit       int       `json:"survivalLimit"`
	DepartingSurvival   int       `json:"departingSurvival"`
	CollectiveCognition int       `json:"collectiveCognition"`
	CurrentYear         int       `json:"currentYear"`
}

File: web/app/lib/types/settlement.ts
import type { UUID } from "node:crypto";

export type Settlement =  SettlementId & {
  survivalLimit: number;
  departingSurvival: number;
  collectiveCognition: number;
  currentYear: number;
};

export type SettlementId = {
  id: UUID;
  name: string;
};